presto-main/src/main/java/com/facebook/presto/cost/ScalarStatsCalculator.java

```diff
@@ -99,5 +105,56 @@ protected SymbolStatsEstimate visitLiteral(Literal node, Void context)
             }
             return estimate.build();
         }
+
+        @Override
+        protected SymbolStatsEstimate visitCast(Cast node, Void context)
+        {
+            SymbolStatsEstimate sourceStats = process(node.getExpression());
+            TypeSignature targetType = TypeSignature.parseTypeSignature(node.getType());
+
+            // todo - make this general postprocessing rule.
+            double distinctValuesCount = sourceStats.getDistinctValuesCount();
+            double lowValue = sourceStats.getLowValue();
+            double highValue = sourceStats.getHighValue();
+
+            if (isIntegralType(targetType)) {
+                // todo handle low/high value changes if range gets narrower due to cast (e.g. BIGINT -> SMALLINT)
+                if (isFinite(lowValue)) {
+                    lowValue = Math.round(lowValue);
+                }
+                if (isFinite(highValue)) {
+                    highValue = Math.round(highValue);
+                }
+                if (isFinite(lowValue) && isFinite(highValue)) {
+                    double integersInRange = highValue - lowValue + 1;
+                    if (!isNaN(distinctValuesCount) && distinctValuesCount > integersInRange) {
```

Comment:
looks like a normalizer code which is missing in `Limit` and `EnforceSingleRow` stats rules

Reply:
Do you want me to do anything here?